src/renderer/shared/api/governance/on-chain/lib/on-chain-utils.ts

```diff
@@ -0,0 +1,123 @@
+import { BN_ZERO, BN } from '@polkadot/util';
+
+import type { ClaimTime, ClaimTimeAt } from './types';
+import { VotingType, VoteType, ReferendumType, Conviction } from '@shared/core';
+import type {
+  Voting,
+  CastingVoting,
+  DelegatingVoting,
+  StandardVote,
+  SplitVote,
+  SplitAbstainVote,
+  ReferendumInfo,
+  OngoingReferendum,
+  RejectedReferendum,
+  ApprovedReferendum,
+  AccountVote,
+} from '@shared/core';
+
+export const onChainUtils = {
+  isCasting,
+  isDelegating,
+  getTotalLock,
+  getLockPeriods,
+
+  isStandardVote,
+  isClaimAt,
+
+  isOngoing,
+  isRejected,
+  isApproved,
+
+  test,
+};
+
+const enum Vote {
+  Nay = 'nay',
+  Aye = 'aye',
+}
+
+function getLockPeriods(conviction: Conviction): number {
+  return {
+    [Conviction.None]: 0,
+    [Conviction.Locked1x]: 1,
+    [Conviction.Locked2x]: 2,
+    [Conviction.Locked3x]: 4,
+    [Conviction.Locked4x]: 8,
+    [Conviction.Locked5x]: 16,
+    [Conviction.Locked6x]: 32,
+  }[conviction];
+}
+
+// Convert aye nay to on-chain values
+function test(vote: Vote, conviction: number): number | undefined {
```

Comment:
I didn't find any use for this function, maybe remove it or rename it?

Reply:
It was a placeholder and will be used during voting with appropriate naming 😌